microsoft/testsuites/vm_extensions/run_command.py

```diff
@@ -0,0 +1,275 @@
+    """,
+)
+class RunCommand(TestSuite):
+    @TestCaseMetadata(
+        description="""
+        Runs the Run Command v2 VM extension with a pre-existing ifconfig script.
+        """,
+        priority=3,
+        requirement=simple_requirement(supported_features=[AzureExtension]),
+    )
+    def verify_existing_script_run(self, log: Logger, node: Node) -> None:
+        settings = {"source": {"CommandId": "ifconfig"}}
+        create_and_verify_extension_run(node, settings)
+
+    @TestCaseMetadata(
+        description="""
+        Runs the Run Command v2 VM extension with a custom shell script.
+        """,
+        priority=3,
+        requirement=simple_requirement(supported_features=[AzureExtension]),
+    )
+    def verify_custom_script_run(self, log: Logger, node: Node) -> None:
+        test_file = f"/tmp/{str(uuid.uuid4())}"
+        settings = {
+            "source": {"CommandId": "RunShellScript", "script": f"touch {test_file}"}
+        }
+        message = f"File {test_file} was not created on the test machine"
+
+        create_and_verify_extension_run(node, settings, f"ls '{test_file}'", 0, message)
+
+    @TestCaseMetadata(
+        description="""
+        Runs the Run Command v2 VM extension with a public uri pointing to the
+        script in blob storage.
+        """,
+        priority=3,
+        requirement=simple_requirement(
+            supported_features=[AzureExtension], supported_platform_type=[AZURE]
+        ),
+    )
+    def verify_public_uri_script_run(
+        self, log: Logger, node: Node, environment: Environment
+    ) -> None:
+        platform = environment.platform
+        assert isinstance(platform, AzurePlatform)
+
+        subscription_id = platform.subscription_id
+        container_name = "rcv2lisa-public"
+        node_context = node.capability.get_extended_runbook(AzureNodeSchema, AZURE)
+        location = node_context.location
+        storage_account_name = get_storage_account_name(
+            subscription_id=subscription_id, location=location
+        )
+        blob_name = "rcv2lisa.sh"
+        test_file = "/tmp/lisatest.txt"
+
+        container_client = get_or_create_storage_container(
+            credential=platform.credential,
+            subscription_id=subscription_id,
+            cloud=platform.cloud,
+            account_name=storage_account_name,
+            container_name=container_name,
+            resource_group_name=AZURE_SHARED_RG_NAME,
+        )
+
+        blob = container_client.get_blob_client(blob_name)
+        if not blob.exists():
+            container_client.set_container_access_policy(
+                signed_identifiers={}, public_access="container"
+            )
+            # Upload blob to container if doesn't exist
+            container_client.upload_blob(
+                name=blob_name, data=f"touch {test_file}"  # type: ignore
+            )
+
+        test_file = "/tmp/lisatest.txt"
```

Comment:
can we move the common code used in verify_public_uri_script_run, verify_private_uri_script_run_failed and verify_sas_uri_script_run into one method?